pkg/envoy/envoy.go

```diff
@@ -80,6 +94,74 @@ func StartEnvoy(debug bool, adminPort int, stateDir, logDir string) *Envoy {
 	return e
 }
 
+// isEOF returns true if the error message ends in "EOF". ReadMsgUnix returns extra info in the beginning.
+func isEOF(err error) bool {
+	strerr := err.Error()
+	errlen := len(strerr)
+	return errlen >= 3 && strerr[errlen-3:] == io.EOF.Error()
+}
+
+// StopEnvoy kills the Envoy process started with StartEnvoy. The gRPC API streams are terminated
+// first.
+func (e *Envoy) StartAccessLogging(accessLogPath string) {
+	// Create the access log listener
+	os.Remove(accessLogPath)
+	var err error
+	e.accessLogListener, err = net.ListenUnix("unixpacket", &net.UnixAddr{Name: accessLogPath, Net: "unixpacket"})
+	if err != nil {
+		log.WithError(err).Fatal("Envoy: Failed to listen at ", accessLogPath)
+	}
+	e.accessLogListener.SetUnlinkOnClose(true)
+
+	go func(uxlis *net.UnixListener) {
+		for {
+			// Each Envoy listener opens a new connection over the Unix domain socket.
+			// Multiple worker threads serving the listener share that same connection
+			uc, err := uxlis.AcceptUnix()
+			if err != nil {
+				log.WithError(err).Error("AcceptUnix failed")
+				continue
+			}
+			log.Info("Envoy: Access log connection opened")
+			go func(conn *net.UnixConn) {
+				buf := make([]byte, 4096)
+				for {
+					n, _, flags, _, err := conn.ReadMsgUnix(buf, nil)
+					if err != nil {
+						if !isEOF(err) {
```

Comment:
`err != io.EOF` doesn't work?

Reply:
No. The error is more specific with "EOF" at the end of the error string.